iceoryx_posh/include/iceoryx_posh/iceoryx_posh_types.hpp

```diff
@@ -97,7 +97,7 @@ constexpr uint32_t MAX_REQUEST_QUEUE_CAPACITY = 1024;
 // Waitset
 namespace popo
 {
-using WaitSetIsConditionSatisfiedCallback = cxx::ConstMethodCallback<bool>;
+using WaitSetIsConditionSatisfiedCallback = cxx::function<bool()>;
```

Comment:
`std::function` and `cxx::function` do not differentiate between

```cpp
bool () const
// and
bool ()

// also not
bool () &&
bool () noexcept
```

One could argue that the reason is tjat die to the copy the original constness isn't valid anymore. However, others want to change `std::function`, see [this proposal](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0045r1.pdf).

Reply:
constness is a concept for methods, i.e. requires objects. The function wrappers have no notion of this given a signature like `bool()`.  The wrappers are not aware of any objects that may or may not change by the call.